### PR TITLE
[dv/otp] Icebox and remove stale todos

### PR DESCRIPTION
**Pull request by cindychip**
This PR adds a TODO to ICEBOX, and remove a stale TODO.

### CODE REVIEW

File: hw/ip/otp_ctrl/dv/env/otp_ctrl_scoreboard.sv [hunk 1]
```diff
@@ -25,8 +25,8 @@ class otp_ctrl_scoreboard #(type CFG_T = otp_ctrl_env_cfg)
   // This bit is used for DAI interface to mark if the read access is valid.
   bit dai_read_valid;
 
-  // TODO: currently skip checking if the ECC error is uncorrectable. Support reading is when we
-  // update to mem_bkdr_if.
+  // ICEBOX(#17798): currently skip checking if the ECC error is uncorrectable.
+  // Support reading is when using mem_bkdr_if to store otp_memory array.
```
Comment (andreaskurth) [inline hw/ip/otp_ctrl/dv/env/otp_ctrl_scoreboard.sv:29]: I don't understand this sentence, sorry. Could you rephrase it?

Comment (cindychip) [reply hw/ip/otp_ctrl/dv/env/otp_ctrl_scoreboard.sv:29]: My bad. I did not put too many comments on the code. I will update that. Thank you for pointing it out.

Comment (andreaskurth) [reply hw/ip/otp_ctrl/dv/env/otp_ctrl_scoreboard.sv]: Thanks, now I understand it :slightly_smiling_face:

**Review by andreaskurth — APPROVED** [review]
